packages/react-reconciler/src/__tests__/ReactHooks-test.internal.js

```diff
@@ -651,11 +654,15 @@ describe('ReactHooks', () => {
     expect(() => {
       expect(() => root.update(<App flip={true} />)).toThrow('custom error');
     }).toWarnDev(
-    // TODO - this should log just once
-    // it logs it twice because of the replay logic in the scheduler I think?
-    // must fix in some way, but pretty edge casey so letting this for now  
+      // TODO - this should log just once
+      // it logs it twice because of the replay logic in the scheduler I think?
+      // must fix in some way, but pretty edge casey so letting this for now
```

Comment:
Why edge casey? Does this happen on every error?

Reply:
The easiest thing we could do is dedupe by message. Keep a DEV-only set of diffs where you emit the warning. Don't emit the same diff twice.